### PR TITLE
Update doctrine/inflector requirement from ^1.3 to ^1.4|^2.0

### PR DESCRIPTION
**Pull request by zingimmick**

### CODE REVIEW

File: rules/code-quality/src/Rector/For_/ForToForeachRector.php [hunk 1]
```diff
@@ -132,14 +138,26 @@ public function refactor(Node $node): ?Node
...
+     * Get the inflector instance.
+     */
+    private static function inflector(): Inflector
+    {
+        if (self::$inflector === null) {
+            self::$inflector = InflectorFactory::create()->build();
+        }
+
+        return self::$inflector;
+    }
```
Comment (TomasVotruba) [inline rules/code-quality/src/Rector/For_/ForToForeachRector.php:159]: We use constructor injection here. The service must be registered in config.php

Comment (zingimmick) [reply rules/code-quality/src/Rector/For_/ForToForeachRector.php]: Fixed

**Comment by TomasVotruba** [comment]
Great job! :+1: 

Could you squash it to 1 commit? So we have relevant git history

**Comment by zingimmick** [comment]
I'm very sorry, I accidentally deleted the branch. recreated #3809 .

**Pull request closed without merging**